### ISSUE TITLE
May I ask what URL to use for google lyrics?

### ISSUE THREAD

**Issue by Logansfury** [Jugran/lyrics-in-terminal#46]
Hello,

I just learned that Google lyrics search is one of the three databases this program uses.

May I ask the URL or the wget that you are using to get lyrics from Google? I asked ChatGPT but it does not seem to know the proper syntax to create a working lyrics request command.

Thank you :)

**Comment by Jugran** [comment]
Following is the configuration for google search

```python
url = 'https://www.google.com/search?q='
HEADER = {
    'User-Agent': 'Mozilla/5.0 (compatible; MSIE 10.0; Windows Phone 8.0; Trident/6.0; IEMobile/10.0; ARM; Touch'
}
CLASS_NAME = r'\w{5,7} \w{4,5} \w{5,7}'  # dependent on User-Agent
```
Append url encoded search query to the above url. Please make sure to use the mentioned user agent.

Also following is the snippet taken from [util.py](https://github.com/Jugran/lyrics-in-terminal/blob/main/lyrics/util.py) for parsing the google search html

```python

def parse_google(html: str) -> List[str] | None:
    ''' parses google result html
        returns list of strings or None if no lyrics found
    '''
    html_regex = re.compile(
        r'<div class="{}">([^>]*?)</div>'.format(CLASS_NAME), re.S)

    text_list = html_regex.findall(html)

    if len(text_list) < 2:
        # No google result found!
        return None

    lyrics_lines = []
    for l in text_list[1:]:
        # lyrics must be multiline,
        # ignore the artist info below lyrics
        if l.count('\n') > 2:
            lyrics_lines += l.split('\n')
    if len(lyrics_lines) < 5:
        # too short match for lyrics
        return None

    return lyrics_lines
```

**Comment by Logansfury** [comment]
Hello Jugran :)

I cannot thank you enough. This was all the code that ChatGPT needed to create a conky application configuration and a bash script using pieces of the above code to successfully get lyrics from the google database. 

I have been running it for about a half hour now and I have yet to find any rare classic rock tune it hasn't been able to provide lyrics to. I am so thrilled to have this adapted from the terminal into an app that gives me all kinds of display options for fonts, colors and images.

Perfect solution, and I am perfectly satisfied, thank you so much for your kindness :)

Have a wonderful holiday season,

Sincerely,

Logan

**Comment by Logansfury** [comment]
I have been able to use ChatGPT resource to construct two conky lyric widgets now, a Google lyrics search, and a combination Google/IRClib search, that checks 2 databases like your Lyrics in Terminal does before resigning itself to displaying an error to finding lyrics.

The Google/IRClib widget still misses a few songs that Lyrics in Terminal displays and I believe this is because I am not also checking Genius database.

I have obtained an API for Genius and have a secret token. May I ask one final question of what is the Genius version of the code snippet you sent me for querying google for lyrics?

Thank you and happy holidays :)
